com.unity.sg2/Editor/GraphUI/DataModel/SGNodeModel.cs

```diff
@@ -80,22 +81,20 @@ public RegistryKey registryKey
         public bool existsInGraphData =>
             m_GraphDataName != null && TryGetNodeHandler(out _);
 
-        protected GraphHandler graphHandler =>
+        GraphHandler graphHandler =>
             ((SGGraphModel)GraphModel).GraphHandler;
 
         ShaderGraphRegistry registry =>
-            ((ShaderGraphStencil)GraphModel.Stencil).GetRegistry();
+            ((SGGraphModel)GraphModel).RegistryInstance;
 
         public bool TryGetNodeHandler(out NodeHandler reader)
         {
             try
             {
-                if (graphDataName == null)
-                {
-                    reader = registry.GetDefaultTopology(m_PreviewRegistryKey);
-                    return true;
-                }
-                reader = graphHandler.GetNode(graphDataName);
+                reader = graphDataName == null ?
+                    registry.GetDefaultTopology(m_PreviewRegistryKey) :
+                    graphHandler.GetNode(graphDataName);
```

Comment:
I know the null case is for read-only operations to setup the searcher. Is there a way we can make this safer? If not, can we add a comment here indicating why we expose default topology to a node model (since there are no other obvious breadcrumbs here that indicate it isn't alive in a graph or could be).